spring-cloud-stream-modules-test-support/src/main/java/org/springframework/cloud/stream/modules/test/PropertiesInitializer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2014 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.modules.test;
+
+import java.util.Properties;
+
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.PropertiesPropertySource;
+
+/**
+ * @author David Turanski
+ */
+public class PropertiesInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
```

Comment:
I don't understand the value of this compared to `[Web]IntegrationTest("foo=bar")`, especially if it is used at class level.
What would be really neat (and to be contributed to boot/core), would be method level Environment (re)setting.